flight/PiOS/posix/pios_queue.c

```diff
@@ -150,7 +150,7 @@ bool PIOS_Queue_Send(struct pios_queue *queuep,
 			if (PIOS_Queue_Send_Impl(queuep, itemp, 1)) {
 				return true;
 			}
-		} while ((timeout_ms != PIOS_QUEUE_TIMEOUT_MAX) &&
+		} while ((timeout_ms == PIOS_QUEUE_TIMEOUT_MAX) ||
```

Comment:
These tests were inverted-- it's important to loop forever if the timeout is MAX, otherwise to use the interval.